### PR TITLE
Enhance create vfe

### PR DESCRIPTION
**Pull request by Kuro2709**
CreateVFE now will allow user to upload nav map as optional. This also include the user can choose the location when they add photosphere. Also fix remove photosphere bug.

### CODE REVIEW

File: src/CreateVFE.tsx [hunk 1]
```diff
@@ -120,16 +169,109 @@ function CreateVFEForm({ onCreateVFE, header }: CreateVFEFormProps) {
...
-          sx={{ margin: "auto" }}
-          variant="contained"
-          onClick={handleCreateVFE}
+          variant="outlined"
+          onClick={() => {
+            setMapDialogOpen(true);
+          }}
         >
-          Create
+          Select Photosphere Location
```
Comment (mnxn) [inline src/CreateVFE.tsx:193]: Maybe this button should be hidden if there is no photosphere?

It's not very useful to open a dialog that only says "No map available".

File: src/buttons/AddPhotosphere.tsx [hunk 2]
```diff
@@ -180,7 +140,19 @@ function AddPhotosphere({
...
       />
-      <PhotosphereCenterFieldset setPhotosphereCenter={setPhotosphereCenter} />
+      <Button
+        variant="outlined"
+        onClick={() => {
+          setMapDialogOpen(true);
+        }}
+      >
+        Select Photosphere Location
+      </Button>
```
Comment (mnxn) [inline src/buttons/AddPhotosphere.tsx:150]: Also, this button can be hidden too when there is no nav map.

File: src/CreateVFE.tsx [hunk 3]
```diff
@@ -120,16 +169,109 @@ function CreateVFEForm({ onCreateVFE, header }: CreateVFEFormProps) {
...
         </Button>
+        {photospherePosition && (
+          <Typography>
+            Selected Center: X: {photospherePosition.x}, Y:{" "}
+            {photospherePosition.y}
+          </Typography>
+        )}
```
Comment (mnxn) [inline src/CreateVFE.tsx:200]: I think this information is unnecessary for the user.

File: src/buttons/AddPhotosphere.tsx [hunk 4]
```diff
@@ -193,6 +165,68 @@ function AddPhotosphere({
           Cancel
         </Button>
       </Stack>
+
+      <Dialog
+        open={mapDialogOpen}
+        onClose={() => {
+          setMapDialogOpen(false);
+        }}
+      >
```
Comment (mnxn) [inline src/buttons/AddPhotosphere.tsx:174]: This dialog and the `handleMapClick` function are repeated here and in `App.tsx`.
It should be abstracted into one component that can be referenced in both locations.

**Review by mnxn — CHANGES_REQUESTED** [review]

File: src/PhotosphereEditor.tsx [hunk 5]
```diff
@@ -171,32 +171,30 @@ function PhotosphereEditor({
...
 
-    const updatedVFE: VFE = {
+    const updatedVFE = {
       ...vfe,
-      photospheres: updatePhotospheres(
-        remainingPhotospheres,
-        photosphereId,
-        null,
-      ),
+      photospheres: remainingPhotospheres,
```
[the remark this answers is not in the document]
Comment (Kuro2709) [reply src/PhotosphereEditor.tsx]: I have changed back to the previous version, the the bug appear again. I think maybe it relate to the upPhotospheres. Do you know how to fix it?

[the remark this answers is not in the document]
Comment (mnxn) [reply src/PhotosphereEditor.tsx]: That's strange. I'm not really sure what the cause of that problem is. Maybe from the `.filter(([key]) => key !== currentPS)` in that function?

Anyways, that can be fixed in another PR, so this PR can be merged without that fix.

**Review by mnxn — APPROVED** [review]

**Pull request merged**